src/twisted/conch/ssh/keys.py

```diff
@@ -863,21 +942,27 @@ def type(self):
         elif isinstance(
                 self._keyObject, (dsa.DSAPublicKey, dsa.DSAPrivateKey)):
             return 'DSA'
+        elif isinstance(
+                self._keyObject, (ec.EllipticCurvePublicKey, ec.EllipticCurvePrivateKey)):
+            return 'EC'
         else:
             raise RuntimeError(
                 'unknown type of object: %r' % (self._keyObject,))
 
 
+
     def sshType(self):
         """
         Get the type of the object we wrap as defined in the SSH protocol,
-        defined in RFC 4253, Section 6.6. Currently this can only be b'ssh-rsa'
-        or b'ssh-dss'.
+        defined in RFC 4253, Section 6.6. Currently this can only be b'ssh-rsa',
+        b'ssh-dss' and ecdsa-sha2-*.
 
         @return: The key type format.
         @rtype: L{bytes}
         """
-        return {'RSA': b'ssh-rsa', 'DSA': b'ssh-dss'}[self.type()]
+
+        return {'RSA': b'ssh-rsa', 'DSA': b'ssh-dss',
+            'EC': b'ecdsa-sha2-nistp' + str(self.size()).encode('utf-8')}[self.type()]
```

Comment:
I wouldn't recommend this.  It makes things less complex, but unnecessarily limits the types of keys supported to the bare minimum.